Shared/Samples/Create and save map/CreateAndSaveMapView.swift

```diff
@@ -0,0 +1,419 @@
+        }
+    }
+}
+
+private extension CreateAndSaveMapView {
+    /// A form that allows the user fill out properties for a map that they
+    /// want to create and save to a portal.
+    struct SaveMapForm: View {
+        /// The portal to save the map to.
+        let portal: Portal
+        
+        /// The folders that the user can save the map to.
+        let folders: [PortalFolder]
+        
+        /// The status of the workflow of creating and saving a map.
+        @Binding var status: Status
+        
+        /// The title of the new map.
+        @State private var title: String = ""
+        
+        /// The tags for the map.
+        @State private var tags: String = ""
+        
+        /// A description of the map.
+        @State private var description: String = ""
+        
+        /// The basemap that the user chose for the new map.
+        @State private var basemap: BasemapOption = .topographic
+        
+        /// The operational data that the user chooses to display on the map.
+        @State private var operationalData: OperationalDataOption = .none
+        
+        /// The folder that the user chose to save the map to.
+        @State private var folder: PortalFolder?
+        
+        /// The map to save to the portal.
+        @State private var map = Map(basemapStyle: BasemapOption.topographic.style)
+        
+        /// The viewpoint of the map view, this will be set as the initial viewpoint
+        /// of the map saved to the portal.
+        @State private var viewpoint: Viewpoint?
+        
+        var body: some View {
+            MapViewReader { mapViewProxy in
+                Form {
+                    Section("Create Map") {
+                        TextField("Title", text: $title)
+                        TextField("Tags", text: $tags)
+                            .autocorrectionDisabled()
+                        TextField("Description", text: $description)
+                        Picker("Folder", selection: $folder) {
+                            ForEach(folders, id: \.self) { folder in
+                                Text(folder.title)
+                                    .tag(folder)
+                            }
+                            Text("None")
+                                .tag(Optional<PortalFolder>.none)
+                        }
+                        Picker("Basemap", selection: $basemap) {
+                            ForEach(BasemapOption.allCases, id: \.self) { value in
+                                Text(value.label)
+                                    .tag(value)
+                            }
+                        }
+                        Picker("Operational Data", selection: $operationalData) {
+                            ForEach(OperationalDataOption.allCases, id: \.self) { value in
+                                Text(value.label)
+                                    .tag(value)
+                            }
+                        }
```

Comment:
```suggestion
                        Picker("Basemap", selection: $basemap) {
                            ForEach(BasemapOption.allCases, id: \.self) { value in
                                Text(value.label)
                            }
                        }
                        Picker("Operational Data", selection: $operationalData) {
                            ForEach(OperationalDataOption.allCases, id: \.self) { value in
                                Text(value.label)
                            }
                        }
```

SwiftUI can actually infer the tags for these two pickers.